File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/ZAIdNumberRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public class ZAIdNumberRule extends LuhnRule {

    public ZAIdNumberRule() {
        super(10);
    }

    @Override
    public boolean isValid(String value) {
        return validate(value, "(\\d){13}");
    }

}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/ZAIdNumberRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

class ZAIdNumberRule : LuhnRule(10) {

    override fun isValid(value: String?): Boolean {
        return value != null && validate(value, "(\\d){13}")
    }

}
